New sample: Line of sight (location)

gdhakal opened the pull request:
This sample demonstrates how to interactively place a line of sight between two locations.

Commit:
Add Line of sight (location) sample

Commit:
Modify comments

Commit:
Modify comments

Commit:
Write self to reference properties

Commit:
Resolve conflict by incorporating both suggestions

Commit:
Change code documentation

Commit:
Change code documentation

Commit:
Change code documentation

nixta reviewed (changes requested):
This is a bit static. Once the observer and target have been set, there is no interactive way to modify the target again.

The demo should really allow the user to drag the target position around and see how the Line of Sight analysis updates in realtime.

This will impact how the instructions UI is set up. @gdhakal I have pulled together a suggestion for how this could work. I'll get with you to see what you think.

nixta committed:
Suggested changes for more interactive LoS demo.

Commit:
Merge pull request #325 from Esri/nf_los_proposal

Suggested changes for more interactive LoS demo.

nixta commented on arcgis-ios-sdk-samples/Analysis/Line of sight (location)/README.md:
```diff
@@ -0,0 +1,15 @@
+
+Once the `AGSLocationLineOfSight` is created, it is added to a collection of analysis overlays in the `AGSSceneView`. The analysis overlays are used to render the results of visual analysis on the scene view.
+
+The sample uses the `geoView(_:didTapAtScreenPoint:mapPoint:)` method on `AGSGeoViewTouchDelegate` to get the tapped point and sets the `observerLocation` on the `AGSLocationLineOfSight`. The `targetLocation` on the `AGSLocationLineOfSight` is updated on `geoView(_:didLongPressAtScreenPoint:mapPoint:)` when user performs long-pressed gesture at a specified location.
```
Oops. I should have included this in my PR. Could we update this text to mention how we use `didMoveLongPress` as well as `didLongPress`?

nixta reviewed (changes requested):
Sorry. I should have thought to update the README.md when I submitted my PR of changes. Could you update it to cover the more interactive approach please?

Commit:
PR review changes

Commit:
Change screenshot since camera position is modified in the sample

gdhakal commented:
@nixta Made changes to the sample. Please take a look. 

- Changed viewpoint with new camera position. The imagery displays well in iPhone 8.
- Changed screenshot `image1.png` as viewpoint is modified in the sample. 
- Documented the added map interaction in README. 
- Removed `self` reference in properties to match with the rest of the code (for consistency purpose).

Commit:
Change README

dg0yal commented:
@gdhakal Code looks good.  Not sure the cyan and magenta colors work for me. I'd much prefer it default to green and red which better conveys what is visible and what isn't. Plus it'll match the other viewshed samples which use a similar color scheme

gdhakal commented:
@dg0yal The cyan and magenta color is used per sample design documentation.

dg0yal commented:
Yeah, i don't think that's the best design decision. The cyan color is commonly used for selecting things, so it appears that part of the line is selected, instead of conveying that that part is visible.

Commit:
PR review changes

gdhakal commented:
@dg0yal The samples now uses default colors to represent visible and obstructed segments. The screenshot and README files are modified accordingly. Please take a look at it.

nixta committed:
Modify to make use of Safe Area for iOS 11 and iPhone X

Commit:
Update README

nixta committed:
Merge pull request #339 from Esri/nf_los_safearea

Modify to make use of Safe Area for iOS 11 and iPhone X

nixta committed:
Merge commit '9739d4cee2b388200c14f6505225f637235dce87' into los

nixta committed:
Fix the sample to avoid an AGSLocationLineOfSight with points at (0,0). Also allow reset of observerLocation with tap.

nixta commented:
@gdhakal Please take a look at my latest pushed changes.

gdhakal commented:
@nixta Thanks Nick. It looks great :shipit:

Commit:
case changes in comments to match with the rest

nixta committed:
Fine-tune a couple of comments.

nixta commented:
Just did the same, @gdhakal. You triggered my OCD.
